src/editor/CodeHintManager.js

```diff
@@ -546,11 +566,20 @@ define(function (require, exports, module) {
                 }
             });
             hintList.onSelect(function (hint) {
-                var restart = sessionProvider.insertHint(hint),
-                    previousEditor = sessionEditor;
-                _endSession();
-                if (restart) {
-                    _beginSession(previousEditor);
+                // allow extensions to handle special hint selections
+                var handled = false;
+                if (hintsAtTopHandler && hintsAtTopHandler.handleHintSelection) {
+                    handled = hintsAtTopHandler.handleHintSelection(hint, sessionEditor, _endSession);
```

Comment:
Match api signatures of related items. for eg: the hints at top handler can have the same signature as normal handler. Eg: `{insertHint, getHints...}`. For Eg. `handled = hintsAtTopHandler.insertHint(...` is eazier to work with in the context of code hints if it does similar behavior.